sql-plugin/src/main/330+/scala/org/apache/spark/RapidsSparkIndexOutOfBoundsException.scala

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark
+
+import ai.rapids.cudf.ColumnVector
+
+object RapidsSparkIndexOutOfBoundsException {
```

Comment:
As I mentioned before, we should move to the v2-style, compile-time shims rather than the v1-style, runtime shims.  That means _not_ adding to the SparkShims interface or touching the *Shims classes.  Instead, there should be an object class under com.nvidia.spark.rapids.shims.v2, something like SparkErrorUtils, where code can just use directly rather than looking up the shim at runtime.  There would be different versions of SparkErrorUtils under sql-plugin based on the Spark version (in this case, one for everything before spark330 and another for 330+).